sklearn/model_selection/tests/test_validation.py

```diff
@@ -769,3 +771,24 @@ def test_cross_val_predict_with_method():
         predictions = cross_val_predict(est, X, y, method=method,
                                         cv=kfold)
         assert_array_almost_equal(expected_predictions, predictions)
+
+
+def test_score_memmap():
+    # Ensure a scalar score of memmap type is accepted
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    clf = MockClassifier()
+    tf = tempfile.NamedTemporaryFile(mode='wb', delete=False)
+    tf.write('Hello world!!!!!')
+    tf.close()
+    scores = np.memmap(tf.name, dtype=float)
+    score = scores.sum()
+    try:
+        cross_val_score(clf, X, y, scoring=lambda est, X, y: score)
+        # non-scalar should still fail
+        assert_raises(ValueError, cross_val_score, clf, X, y,
+                      scoring=lambda est, X, y: scores)
+    except Exception:
```

Comment:
I don't think that you need the except clause. You could be doing only the try/finally.